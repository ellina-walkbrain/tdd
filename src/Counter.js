import React, { useCallback, useState } from "react";

export default function Counter() {
    const [count, setCount] = useState(0);

    const onClickPlus = useCallback(() => {
        setCount(count + 1);
    }, [count]);

    const onClickMinus = useCallback(() => {
        setCount(count - 1);
    }, [count]);

    return (
        <div className="counter">
            <h1>counter입니다</h1>
            <button className="plus-button" onClick={onClickPlus}>
                +
            </button>
            <span>{count}</span>
            <button className="plus-button" onClick={onClickMinus}>
                -
            </button>
        </div>
    );
}
