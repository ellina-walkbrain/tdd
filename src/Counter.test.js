import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Counter from "./Counter";

describe("<Counter/>", () => {
    it("matches snapshot", () => {
        const utils = render(<Counter />);
        expect(utils.container).toMatchSnapshot();
    });
    it("has number and two buttons", () => {
        //0과 + - 있는지 확인
        const utils = render(<Counter />);
        expect(utils.container).toMatchSnapshot();
        utils.getByText("0");
        utils.getByText("+");
        utils.getByText("-");
    });

    it("increases", () => {
        //증가
        const utils = render(<Counter />);
        const number = utils.getByText("0");
        const plusButton = utils.getByText("+");
        // 클릭 이벤트를 두번 발생시키기
        fireEvent.click(plusButton);
        fireEvent.click(plusButton);

        //두번 발생시켰을때 2가 되는지 확인
        expect(number.textContent).toBe("2");
    });

    it("decreases", () => {
        const utils = render(<Counter />);
        const number = utils.getByText("0");
        const plusButton = utils.getByText("-");
        // 클릭 이벤트를 두번 발생시키기
        fireEvent.click(plusButton);
        fireEvent.click(plusButton);
        expect(number).toHaveTextContent("-2"); // jest-dom 의 확장 matcher 사용
    });
});
