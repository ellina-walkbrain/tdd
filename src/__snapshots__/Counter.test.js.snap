// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> has number and two buttons 1`] = `
<div>
  <div
    class="counter"
  >
    <h1>
      counter입니다
    </h1>
    <button
      class="plus-button"
    >
      +
    </button>
    <span>
      0
    </span>
    <button
      class="plus-button"
    >
      -
    </button>
  </div>
</div>
`;

exports[`<Counter/> matches snapshot 1`] = `
<div>
  <div
    class="counter"
  >
    <h1>
      counter입니다
    </h1>
    <button
      class="plus-button"
    >
      +
    </button>
    <span>
      0
    </span>
    <button
      class="plus-button"
    >
      -
    </button>
  </div>
</div>
`;
